cli/commands/benchmark-web-vitals.mjs

```diff
@@ -164,7 +164,7 @@ export async function handler( opt ) {
 	for await ( const url of getURLs( opt ) ) {
 		const { completeRequests, metrics } = await benchmarkURL(
 			browser,
-			params
+			params // TODO: This should be passing the url and not params.url.
```

Comment:
Can you clarify why? Further down in `benchmarkURL()` you added a comment that the viewport should be configurable, so I'd argue that alone would be a good reason to keep using `params`.

Reply:
I added this because `params` contains either `url` or `file`. The list of URLs is obtained from `getURLs()`. It doesn't seem to be accounting for the case where the URLs are supplied via `file`. It is assuming that `getURLs( opt )[0]` is always the same as `params.url`, as I understand.

Reply:
For example, if I do this:

```
npm run research benchmark-web-vitals -- --file urls.txt 
```

Where `urls.txt` contains:

```
https://wordpress.org/
https://make.wordpress.org/
https://wptavern.com/
```

I get an error:

```
> research
> ./cli/run.mjs benchmark-web-vitals --file urls.txt

TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:405:5)
    at new URL (node:internal/url:637:13)
    at benchmarkURL (file:///home/westonruter/repos/wpp-research/cli/commands/benchmark-web-vitals.mjs:253:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async handler (file:///home/westonruter/repos/wpp-research/cli/commands/benchmark-web-vitals.mjs:165:41)
    at async Command.<anonymous> (file:///home/westonruter/repos/wpp-research/cli/run.mjs:65:4) {
  input: 'undefined',
  code: 'ERR_INVALID_URL'
}
```

This is is erroring here:

```js
const url = new URL( params.url );
```

Because the URLs aren't being provided via `--url` but rather via `--file`.

Reply:
Fixed in 075bb5f:

```
$ npm run --silent research benchmark-web-vitals -- --file urls.txt 
╔═══════════════════╤════════════════════════╤═════════════════════════════╤═══════════════════════╗
║ URL               │ https://wordpress.org/ │ https://make.wordpress.org/ │ https://wptavern.com/ ║
╟───────────────────┼────────────────────────┼─────────────────────────────┼───────────────────────╢
║ Success Rate      │ 100%                   │ 100%                        │ 100%                  ║
╟───────────────────┼────────────────────────┼─────────────────────────────┼───────────────────────╢
║ FCP (median)      │ 1081.7                 │ 806.1                       │ 1515.1                ║
╟───────────────────┼────────────────────────┼─────────────────────────────┼───────────────────────╢
║ LCP (median)      │ 1081.7                 │ 806.1                       │ 1523.4                ║
╟───────────────────┼────────────────────────┼─────────────────────────────┼───────────────────────╢
║ TTFB (median)     │ 626.7                  │ 470.9                       │ 997.3                 ║
╟───────────────────┼────────────────────────┼─────────────────────────────┼───────────────────────╢
║ LCP-TTFB (median) │ 455                    │ 335.2                       │ 526.1                 ║
╚═══════════════════╧════════════════════════╧═════════════════════════════╧═══════════════════════╝
```

Reply:
Thanks, now I get it. Great catch!